File: test/ch21-benchmarks/bench-007-fibonacci.go
// BENCH-007: Fibonacci recursive (n=20) - Go
package main

func fibonacci(n int) int {
	if n <= 1 {
		return n
	}
	return fibonacci(n-1) + fibonacci(n-2)
}

func main() {
	result := fibonacci(20)
	_ = result
	// Silent for benchmarking
	// Expected: 6765
}

File: test/ch21-benchmarks/bench-007-fibonacci.jl
# BENCH-007: Fibonacci recursive (n=20) - Julia
function fibonacci(n::Int64)::Int64
    if n <= 1
        return n
    else
        return fibonacci(n - 1) + fibonacci(n - 2)
    end
end

function main()
    result = fibonacci(20)
    # Silent for benchmarking
    # Expected: 6765
end

main()
